### ISSUE TITLE
Invariant error when deleting design from project

### ISSUE THREAD

**Issue by Abaddonica** [avocode/avocode#477]
Made an attempt to delete design from project.

-----
App v.2.4.3-3630f5a
Source UID: 99597
```
Error: Minified exception occurred; use the non-minified dev environment for the full error message and additional helpful warnings.
    at invariant (/home/yut/Avocode/resources/app.asar/node_modules/flux/node_modules/fbjs/lib/invariant.js:35:15)
    at o.dispatch (/home/yut/Avocode/resources/app.asar/node_modules/flux/lib/Dispatcher.js:167:160)
    at o.handleServerAction (/home/yut/Avocode/resources/app.asar/node_modules/@avocode/core/lib/dispatcher.js:1:980)
    at e._emitItemProcessingProgress (/home/yut/Avocode/resources/app.asar/node_modules/@avocode/core/lib/uploader.js:1:10574)
    at /home/yut/Avocode/resources/app.asar/node_modules/@avocode/core/lib/uploader.js:1:7483
    at Request._callback (/home/yut/Avocode/resources/app.asar/node_modules/@avocode/core/lib/api.js:1:8607)
    at Request.self.callback (/home/yut/Avocode/resources/app.asar/node_modules/@avocode/core/node_modules/request/request.js:198:22)
    at emitTwo (events.js:92:20)
    at Request.emit (events.js:169:7)
    at Request.<anonymous> (/home/yut/Avocode/resources/app.asar/node_modules/@avocode/core/node_modules/request/request.js:1063:14)
    at emitOne (events.js:82:20)
    at Request.emit (events.js:166:7)
    at IncomingMessage.<anonymous> (/home/yut/Avocode/resources/app.asar/node_modules/@avocode/core/node_modules/request/request.js:1009:12)
    at emitNone (events.js:72:20)
    at IncomingMessage.emit (events.js:163:7)
    at _stream_readable.js:891:16
    at process._tickCallback (node.js:357:13)
    at EventEmitter.ipc.sendSync (/home/yut/Avocode/resources/atom.asar/renderer/api/lib/ipc.js:21:31)
    at Object.RemoteMemberFunction [as showMessageBox] (/home/yut/Avocode/resources/atom.asar/renderer/api/lib/remote.js:110:29)
    at window.confirm (/home/yut/Avocode/resources/atom.asar/
```

**Comment by rychlis** [comment]
Thanks for the report, are you able to reproduce this error every time you try to delete a design or did it happen just once randomly?

**Comment by rychlis** [comment]
OK we now know the cause most probably, we should be able to fix this in the next release

**Comment by rychlis** [comment]
This error might happen when you try to delete a design that you are currently uploading, was that the case?

**Comment by Abaddonica** [comment]
I've tried to open .psd template and began uploading. But process have been stuck on 50% and I've tried to restart Avocode.
After restart I've found Avocode unable to upload or delete (this issue is about that deletion) this template and uploaded all necessary files via website. Only after that I've became able top work with templates.
I cannot reproduce bug in near future because license for our company have ended and my boss is not going to renew it soon.
